### ISSUE TITLE
Migration Error with PostgreSQL when upgrading to v2.0

### ISSUE THREAD

**Issue by crgwbr** [pinax/django-mailer#115]
Exception:

```
django.db.utils.ProgrammingError: operator class "varchar_pattern_ops" does not accept data type smallint
```

Log lines from DB:

```
ERROR:  operator class "varchar_pattern_ops" does not accept data type smallint
postgres_1  | 2019-09-23 16:36:12.462 UTC [33] STATEMENT:  ALTER TABLE "mailer_messagelog" ALTER COLUMN "priority" TYPE smallint USING "priority"::smallint
```

Similar issue from Stack Overflow: https://stackoverflow.com/questions/20284557/postgresql-operator-class-varchar-pattern-ops-does-not-accept-data-type-intege

It seems like the existing index on the `mailer_messagelog.priority` is causing the issuing. Dropping the index before changing the column type, then recreating the index might resolve it.

**Comment by spookylukey** [comment]
Can you describe when this happens? Have the migrations been run on the DB?

**Comment by crgwbr** [comment]
In our app this is happening in the unit test suite; specifically, while Django is running migrations to prepare the DB for the test suite. We also know it's specific to v2.0 since the same setup worked fine for us with the last version.

**Comment by spookylukey** [comment]
I'm guessing this must be specific to Postgres version, because I haven't experienced it with my own Postgres instances. Could you produce a migration that does what you suggested and works for your case? It's a bit tricky because of other people who may have successfully run the migration. So I think it needs to go as a change to the existing migration.

**Comment by bastbnl** [comment]
Same here. I'm using `psql (PostgreSQL) 11.5 (Debian 11.5-1.pgdg90+1)` on Debian `Linux 4.9.0-7-amd64 #1 SMP Debian 4.9.110-1 (2018-07-05) x86_64 GNU/Linux`. 

I was able to install the migration after dropping both indexes:
```
drop index "mailer_messagelog_priority";
drop index "mailer_messagelog_priority_like";
```

I can't tell if the migration would've ran after deleting only the `mailer_messagelog_priority_like` index, as I had already deleted the first and deleting just that index was insufficient.
